app/javascript/components/remove-catalog-item-modal.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Modal } from 'patternfly-react';
+
+const parseApiError = (error) => {
+  const { data: { error: { message } } } = error;
+  return message;
+};
+
+const removeCatalogItems = (catalogItems) => {
+  let apiPromises = [];
+
+  miqSparkleOn();
+  catalogItems.forEach(item => {
+    apiPromises.push(API.post(`/api/service_templates/${item.id}`, {action: 'delete'}, {skipErrors: [400, 500]})
+                       .then((apiResult) => ({result: 'success', data: apiResult, name: item.name}))
+                       .catch((apiResult) => ({result: 'error', data: apiResult, name: item.name})))
+  });
+  Promise.all(apiPromises)
+    .then((apiData) => {
+      if (catalogItems.length > 1 || (catalogItems.length === 1 && apiData[0].result === 'success')) {
```

Comment:
When there's more than 1 report, regardless of result, we redirect.
So, the error message is gone.

But there's code on line 35 trying to deal with that case and show a flash message.

... This *could* work in this order, but likely won't at least in some cases.

I think all the `miqFlashLater` need to happen *before* assignment to `window.location.href`

Reply:
Fixed.